### PR TITLE
convert ConversationContentViewController

### PR DESCRIPTION
**Pull request by billypchan**
## What's new in this PR?

convert `ConversationContentViewController` to Swift

### CODE REVIEW

File: Wire-iOS/Sources/UserInterface/Conversation/Content/ConversationContentViewController.swift [hunk 1]
```diff
@@ -17,27 +17,143 @@
...
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    deinit {
+        // Observer must be deallocated before `mediaPlaybackManager`
+        activeMediaPlayerObserver = nil
+        mediaPlaybackManager = nil
+
+            tableView.delegate = nil
```
Comment (typfel) [inline Wire-iOS/Sources/UserInterface/Conversation/Content/ConversationContentViewController.swift]: Extra indentation

Comment (billypchan) [reply Wire-iOS/Sources/UserInterface/Conversation/Content/ConversationContentViewController.swift]: The `deinit` is no longer necessary and it is removed.

File: Wire-iOS/Sources/UserInterface/Conversation/Content/ConversationContentViewController.swift [hunk 2]
```diff
@@ -17,27 +17,128 @@
...
+        tableView.estimatedRowHeight = 80
+        tableView.translatesAutoresizingMaskIntoConstraints = false
+        tableView.allowsSelection = true
+        tableView.allowsMultipleSelection = false
+        tableView.delegate = self
+        tableView.separatorStyle = .none
+        tableView.delaysContentTouches = false
+        tableView.keyboardDismissMode = AutomationHelper.sharedHelper.disableInteractiveKeyboardDismissal ? .none : .interactive
+
+            tableView.backgroundColor = UIColor.from(scheme: .contentBackground)
```
Comment (typfel) [inline Wire-iOS/Sources/UserInterface/Conversation/Content/ConversationContentViewController.swift]: Extra indentation. `performWithoutAnimation*(` wasn't necessary anymore?

File: Wire-iOS/Sources/UserInterface/Conversation/Content/ConversationContentViewController.swift [hunk 3]
```diff
@@ -17,27 +17,128 @@
...
+            bottomContainer.bottomAnchor.constraint(equalTo: view.bottomAnchor)
+            ])
+        let heightCollapsingConstraint = bottomContainer.heightAnchor.constraint(equalToConstant: 0)
+        heightCollapsingConstraint.priority = .defaultHigh
+        heightCollapsingConstraint.isActive = true
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
```
Comment (typfel) [inline Wire-iOS/Sources/UserInterface/Conversation/Content/ConversationContentViewController.swift]: We were previously setting the dataSource here, how does that work now?

File: Wire-iOS/Sources/UserInterface/Conversation/Content/ConversationContentViewController.swift (hunk 2 above)
Comment (billypchan) [reply Wire-iOS/Sources/UserInterface/Conversation/Content/ConversationContentViewController.swift]: Tested on simulator and it does not have any issue without `performWithoutAnimation`

File: Wire-iOS/Sources/UserInterface/Conversation/Content/ConversationContentViewController.swift [hunk 4]
```diff
@@ -55,44 +156,217 @@ extension ConversationContentViewController {
...
+    func setTableViewBottomMargin(_ bottomMargin: CGFloat) {
+        var insets = tableView.correctedContentInset
+        insets.bottom = bottomMargin
+        tableView.correctedContentInset = insets
+        tableView.contentOffset = CGPoint(x: tableView.contentOffset.x, y: -bottomMargin)
+    }
+
+    var isScrolledToBottom: Bool {
+        return dataSource.hasNewerMessagesToLoad == nil && tableView.contentOffset.y + tableView.correctedContentInset.bottom <= 0
```
Comment (typfel) [inline Wire-iOS/Sources/UserInterface/Conversation/Content/ConversationContentViewController.swift]: Isn't `hasNewerMessagesToLoad` a boolean? it was previously `!self.dataSource.hasNewerMessagesToLoad`

File: Wire-iOS/Sources/UserInterface/Conversation/Content/ConversationContentViewController.swift (hunk 3 above)
Comment (billypchan) [reply Wire-iOS/Sources/UserInterface/Conversation/Content/ConversationContentViewController.swift]: It is created with `lazy var` now to get rid of optional.

File: Wire-iOS/Sources/UserInterface/Conversation/Content/ConversationContentViewController.swift (hunk 4 above)
Comment (billypchan) [reply Wire-iOS/Sources/UserInterface/Conversation/Content/ConversationContentViewController.swift]: fixed

**Pull request merged**